editor/sidebar/post-sticky/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { FormToggle, withInstanceId } from 'components';
+
+/**
+ * Internal dependencies
+ */
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+function PostSticky( { postSticky = false, instanceId, ...props } ) {
+	const stickyToggleId = 'post-sticky-toggle-' + instanceId;
+
+	const onTogglePostSticky = () => props.editPost( { sticky: ! postSticky } );
```

Comment:
Minor: I'd personally bake this in the `mapStateToDispatch` providing a `onTogglePostSticky` prop instead that way the `PostSticky` presentational component would deal only with the sticky value and a simple callback. Doesn't need to know about `editPost`.